code/mob/dead/observer.dm

```diff
@@ -101,14 +101,45 @@
 	if (!P.AH)
 		return
 
+	var/is_mutantrace = FALSE
+	if ("lizard" in P.traitPreferences.traits_selected)
+		src.icon_state = "ghost-lizard"
+		is_mutantrace = TRUE
+	else if ("cow" in P.traitPreferences.traits_selected)
+		src.icon_state = "ghost-cow"
+		is_mutantrace = TRUE
+	else if ("skeleton" in P.traitPreferences.traits_selected)
+		src.icon_state = "ghost-skeleton"
+		is_mutantrace = TRUE
+	else if ("roach" in P.traitPreferences.traits_selected)
+		src.icon_state = "ghost-roach"
+		is_mutantrace = TRUE
+	else if ("pug" in P.traitPreferences.traits_selected)
+		src.icon_state = "ghost-pug"
+		is_mutantrace = TRUE
```

Comment:
Shouldn't this just be a var on the mutantrace?

Reply:
This is from client, not mob mutantrace var

Reply:
You can just get the mutantrace datum from their trait list though, surely?

Reply:
Sorry, I'm still not understanding

If you are a /mob/new_player and start, mutantrace and hair_override vars won't be accessible?

Reply:
They'll have the trait in their trait list, which contains a reference to the mutantrace type.